staging/src/k8s.io/component-helpers/resource/helpers.go

```diff
@@ -59,18 +61,19 @@ func PodRequests(pod *v1.Pod, opts PodResourcesOptions) v1.ResourceList {
 		for i := range pod.Status.ContainerStatuses {
 			containerStatuses[pod.Status.ContainerStatuses[i].Name] = &pod.Status.ContainerStatuses[i]
 		}
+		if opts.IsSidecarContainersEnabled {
+			for i := range pod.Status.InitContainerStatuses {
+				containerStatuses[pod.Status.InitContainerStatuses[i].Name] = &pod.Status.InitContainerStatuses[i]
+			}
+		}
 	}
 
 	for _, container := range pod.Spec.Containers {
 		containerReqs := container.Resources.Requests
 		if opts.InPlacePodVerticalScalingEnabled {
 			cs, found := containerStatuses[container.Name]
 			if found {
-				if pod.Status.Resize == v1.PodResizeStatusInfeasible {
-					containerReqs = cs.AllocatedResources.DeepCopy()
-				} else {
-					containerReqs = max(container.Resources.Requests, cs.AllocatedResources)
-				}
+				containerReqs = setContainerReqs(pod, &container, cs)
```

Comment:
you added init containers to this collection, but this loop only iterates the regular containers. Is it intended?

Reply:
Yes this is intended since init containers are added in the below code starting from line 98.